test-utils/src/random.rs

```diff
@@ -60,3 +60,51 @@ impl From<u64> for Seed {
 pub fn make_seedable_rng(seed: Seed) -> impl Rng + CryptoRng {
     ChaChaRng::seed_from_u64(seed.0)
 }
+
+// This is similar to SliceRandom::shuffle, but it makes sure that the resulting order
+// will be different from the original one.
+pub fn shuffle_until_different<T>(slice: &mut [T], rng: &mut impl Rng) {
+    const MAX_ATTEMPTS: usize = 1000;
+
+    for _ in 0..MAX_ATTEMPTS {
+        let mut swapped = false;
+
+        for i in (1..slice.len()).rev() {
+            let other_idx = rng.gen_range(0..(i + 1));
+            if other_idx != i {
+                slice.swap(i, other_idx);
```

Comment:
There's a method shuffle() from the SliceRandom trait, I believe...

Reply:
Yes, but it may leave the collection unchanged as well. This variant ensures that the result is always different from the original.

Reply:
If all you wanted to prove is that you're sending one block that isn't the first one, isn't easier to just swap the first element with a random element of the rest?

I don't really care that much, but this looks like hitting a fly with a bazooka. You can keep it, I don't mind.